src/server/internalServer.cpp

```diff
@@ -744,6 +747,41 @@ std::unique_ptr<Response> InternalServer::handle_viewer_settings(const RequestCo
   return ContentResponse::build(*this, RESOURCE::templates::viewer_settings_js, data, "application/javascript; charset=utf-8");
 }
 
+std::unique_ptr<Response> InternalServer::handle_no_js(const RequestContext& request)
+{
+
+  const auto url = request.get_url();
+  const auto domains = kiwix::split(url, "/", true, false);
+  auto filter = get_search_filter(request);
+  try {
+    if (request.get_argument("category") == "") {
+      filter.clearCategory();
+    }
+    if (request.get_argument("lang") == "") {
+      filter.clearLang();
+    }
+  } catch (...) {}
+  OPDSDumper opdsDumper(mp_library, mp_nameMapper);
+  opdsDumper.setRootLocation(m_root);
+  opdsDumper.setLibraryId(getLibraryId());
+  std::string content = opdsDumper.noJSData(filter);
+
+  if (domains.size() == 1) {
+    content = opdsDumper.noJSData(filter);
+  } else if ((domains.size() == 3) && (domains[1] == "download") && (mp_nameMapper->getIdForName(domains[2]) != "")) {
+    content = opdsDumper.noJSDownloadLinks(mp_nameMapper->getIdForName(domains[2]));
+  } else {
+    return HTTP404Response(*this, request)
+           + urlNotFoundMsg;
+  }
```

Comment:
This set the content to the result of `opdsDumper.noJSData(filter)` all the time and then overwrite it.
So the first initialization is useless. And we should create the filter only if `domain.size() == 1`, not all the time.